src/device-manager/java/src/nl/Weave/DataManagement/GenericTraitUpdatableDataSinkImpl.java

```diff
@@ -0,0 +1,358 @@
+    {
+        setSigned(path, value, false);
+    }
+
+    @Override
+    public void setSigned(String path, BigInteger value)
+    {
+        setSigned(path, value, false);
+    }
+
+    @Override
+    public void setUnsigned(String path, int value)
+    {
+        setUnsigned(path, value, false);
+    }
+
+    @Override
+    public void setUnsigned(String path, long value)
+    {
+        setUnsigned(path, value, false);
+    }
+
+    @Override
+    public void setUnsigned(String path, BigInteger value)
+    {
+        setUnsigned(path, value, false);
+    }
+
+    @Override
+    public void set(String path, double value)
+    {
+        set(path, value, false);
+    }
+
+    @Override
+    public void set(String path, boolean value)
+    {
+        set(path, value, false);
+    }
+
+    @Override
+    public void set(String path, String value)
+    {
+        set(path, value, false);
+    }
+
+    @Override
+    public void set(String path, byte[] value)
+    {
+        set(path, value, false);
+    }
+
+    @Override
+    public void setNull(String path)
+    {
+        setNull(path, false);
+    }
+
+    @Override
+    public int getInt(String path)
+    {
+        ensureNotClosed();
+        return (int)getLong(mTraitInstancePtr, path);
+    }
+
+    @Override
+    public long getLong(String path)
+    {
+        ensureNotClosed();
+        return getLong(mTraitInstancePtr, path);
+    }
+
+    @Override
+    public BigInteger getBigInteger(String path)
+    {
+        ensureNotClosed();
+
+        long value = getLong(mTraitInstancePtr, path);
+        BigInteger bigInt = BigInteger.valueOf(value);
+        return bigInt;
+    }
+
+    @Override
+    public double getDouble(String path)
+    {
+        ensureNotClosed();
+        return getDouble(mTraitInstancePtr, path);
+    }
+
+    @Override
+    public boolean getBoolean(String path)
+    {
+        ensureNotClosed();
+        return getBoolean(mTraitInstancePtr, path);
+    }
+
+    @Override
+    public String getString(String path)
+    {
+        ensureNotClosed();
+        return getString(mTraitInstancePtr, path);
+    }
+
+    @Override
+    public byte[] getBytes(String path)
+    {
+        ensureNotClosed();
+        return getBytes(mTraitInstancePtr, path);
+    }
+
+    @Override
+    public long getVersion()
+    {
+        ensureNotClosed();
+        return getVersion(mTraitInstancePtr);
+    }
+
+    @Override
+    public void beginRefreshData()
+    {
+        ensureNotClosed();
+        beginRefreshData(mTraitInstancePtr);
+    }
+
+    @Override
+    public CompletionHandler getCompletionHandler()
+    {
+        return mCompHandler;
+    }
+
+    @Override
+    public void setCompletionHandler(CompletionHandler compHandler)
+    {
+        mCompHandler = compHandler;
+    }
+
+    private void onError(Throwable err)
+    {
+        if (mCompHandler == null)
+        {
+            throw new IllegalStateException("unexpected err, mCompHandler is null");
```

Comment:
Since this will be surfaced to the client, consider an error such as:

"No CompletionHandler is set. Did you forget to call setCompletionHandler()?"

Same for onRefreshdataComplete()

Reply:
just update the error message, done, thanks